### ISSUE TITLE
rate4site error

### ISSUE THREAD

**Issue by afrubin** [kc-lab/dms2dfe#4]
I encountered the following error when running `dms2dfe Olson_et_al_2014` from the Overall Testing section of [Example Datasets](http://kc-lab.github.io/dms2dfe/v1.0.6/html/6test.html).

```
[2017-09-25 09:48:57,202] ERROR	from io_data_files.py in info2src(..): Path to files do not exist. Include correct path in cfg/info. rate4site_fh : /home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/rate4site/rate4site
rate4site_fh: Optional: path to rate4site source file (for calculation of conservation scores) (default: nan) =
```

It seems that there's an extra `rate4site` at the end of the file path, although the path in `cfg/info` appears to be correct:

```
rate4site_fh,/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/rate4site/,,,Optional: path to rate4site source file (for calculation of conservation scores)
```

If I enter in the path `/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/rate4site/` at the prompt, I get the error `IOError: [Errno 2] No such file or directory: '/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/rate4site/.rate4site'` (similar if I enter the path `/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/`).

In addition, there's a problem with `bowtie2` not being found. The specified directory `/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/bowtie2-2.2.1/bowtie2-build` is empty and the `dms2dfe` throws an `OSError` before continuing. See the attached output for details: [rate4site.txt](https://github.com/kc-lab/dms2dfe/files/1327889/rate4site.txt)

**Comment by kc-lab** [comment]
I would guess the errors with dependencies are likely because the of clean
VM being used. Basic bash command to download stuff i.e. `wget` may not be
available on the VM.

Following things can be done.
1. Please send me the /home/afrubin/anaconda3/envs/
dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/ folder.
It would have log (*.log) files.
2. Let me know how the VM being used is made. So I can recreate it myself
and check this error beforehand.
3. may be I would add wget in the requirements list.

On 25 Sep 2017 5:42 am, "Alan Rubin" <notifications@github.com> wrote:

I encountered the following error when running dms2dfe Olson_et_al_2014
from the Overall Testing section of Example Datasets
<http://kc-lab.github.io/dms2dfe/v1.0.6/html/6test.html>.

[2017-09-25 09:48:57,202] ERROR	from io_data_files.py in info2src(..):
Path to files do not exist. Include correct path in cfg/info.
rate4site_fh : /home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/rate4site/rate4site
rate4site_fh: Optional: path to rate4site source file (for calculation
of conservation scores) (default: nan) =

It seems that there's an extra rate4site at the end of the file path,
although the path in cfg/info appears to be correct:

rate4site_fh,/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/rate4site/,,,Optional:
path to rate4site source file (for calculation of conservation scores)

If I enter in the path /home/afrubin/anaconda3/envs/
dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/
rate4site-3.0.0/src/rate4site/ at the prompt, I get the error IOError:
[Errno 2] No such file or directory: '/home/afrubin/anaconda3/envs/
dms2dfe/lib/python2.7/site-packages/dms2dfe/dms2dfe_dependencies/rate4site/
rate4site-3.0.0/src/rate4site/.rate4site' (similar if I enter the path
/home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-
packages/dms2dfe/dms2dfe_dependencies/rate4site/rate4site-3.0.0/src/).

In addition, there's a problem with bowtie2 not being found. The specified
directory /home/afrubin/anaconda3/envs/dms2dfe/lib/python2.7/site-
packages/dms2dfe/dms2dfe_dependencies/bowtie2-2.2.1/bowtie2-build is empty
and the dms2dfe throws an OSError before continuing. See the attached
output for details: rate4site.txt
<https://github.com/kc-lab/dms2dfe/files/1327889/rate4site.txt>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/kc-lab/dms2dfe/issues/4>, or mute the thread

**Comment by afrubin** [comment]
The VM was created using VirtualBox and a standard installation of Debian 9 Jesse.

Having looked at the logs in the dependencies directory as suggested, you are correct that the issue was caused by some missing basic dependencies (specifically `make` and `g++`).

While it is reasonable that the installation process fails when these are not present, the program should check the exit status of the dependency build steps and report an error rather than fail silently.

Installing `make` and `g++` seem to have fixed the problem with `bowtie2` but there's still a compiler error when building `rate4site`. Here's the dependencies log file: [dms2dfe_dependencies.log](https://github.com/kc-lab/dms2dfe/files/1327953/dms2dfe_dependencies.log)

**Comment by rraadd88** [comment]
I have now tested the installation on Debian 9 'Stretch' (64bit) and resolved the aforementioned compilation errors. The updated version is now 1.0.6.4 .

Apparently, some basic dependencies are absolutely essential for the installation. This is especially a problem with Debians (unlike Ubuntu). I have [recommended installation of the basic dependencies](http://kc-lab.github.io/dms2dfe/v1.0.6/html/1installation.html#installation) before actual install of dms2dfe.

Instead of failing silently, now, I have incorporated error reporting and some interactive steps that would ensure the installation of dependencies is carried out. For example, if the installation of 'rate4site' is failed (weirdly, it is always the case with debian system), the error is caught and user is advised to install the program through APT instead.

Hope this resolves this issue. also, very sorry for the long delay. For some unavoidable circumstances, I was away from my workplace for a month.
